src/qutip_cupy/__init__.py

```diff
@@ -38,6 +38,14 @@
 )
 data.to.register_aliases(["cupyd"], CuPyDense)
 
+
+def is_cupydense(data):
+    return isinstance(data, CuPyDense)
```

Comment:
I think that checking whether the class is CuPyDense is not necessary as the dispatcher does that by default (at least I did not have to do it in qutip-tensorflow). Testing whether is `cupy.array` (or whatever the array class in CuPy is) is useful as the Qobj will default to your constructor in that case.